### ISSUE TITLE
Openshotv2.5.1-dev2 blender-2.82a crash "can't use animated title"

### ISSUE THREAD

**Issue by SebO-L** [OpenShot/openshot-qt#3408]
**Describe the bug**
OpenShot-v2.5.1-dev2-1587624156-15c3efb7-26b7db94-x86_64.AppImage blender-2.82a-linux64 AnimatedTitle crash
Can't use blender for animated Titled with the last version of OpenShot
Generate button is greyed
I use the daily build of Openshot because the last stable build is not working on my debian machine.
I download the last blender binary (2.82a) from the official site because the version in the stable debian package manager is too old (2.79)
I reproduced the same bug on Ubuntu 18.04 with the same package.  

**Steps to reproduce the behavior:**
1. Select the blender binary in Preferences
2. Go to Title-> Animated Title
3. Select one of the title "snow effect for example"
4. The generate button is greyed

**Expected behavior**
Generation of my animated title with blender

**System Details**
- OpenShot Version: 2.5.1-dev2
- libopenshot Version:  0.2.5-dev2
- Platform: Linux-4.19.0-8-amd64-x86_64-with-debian-10.3
- Processor: 
- Machine: x86_64
- Python version: 3.4.3
- Qt5 version: 5.2.1
- PyQt5 version: 5.2.1
- Qt Detected Languages: ['fr-FR']
- LANG Environment Variable: fr_FR.UTF-8
- LOCALE Environment Variable: 
- Daily Build Installed: http://github.com/OpenShot/openshot-qt/releases/download/OpenShot-v2.5.1-dev2-1587624156-15c3efb7-26b7db94-x86_64.AppImage

**Log Files**
- http://logs.openshot.org/k2bi7jrvpni9eqk02t470kscf4d5o9yt/openshot-qt.zip
- http://logs.openshot.org/k2bi7jrvpni9eqk02t470kscf4d5o9yt/libopenshot.zip

**Exception / Stacktrace**
```No stacktrace found in log files```

**Comment by ferdnyc** [comment]
Gah, seems `subprocess` is unhappy with Blender printing non-ASCII pathnames, when the default encoding is detected as 'ascii'`. I believe #3413 will fix this even for Python 3.4.

**Comment by ferdnyc** [comment]
@SebO-L Here's a link to an AppImage built with #3413 applied:

https://drive.google.com/open?id=1LIXaUEezQjiganUjr-DIvkn1Wvfc5gxi

Could you test whether that solves this issue for you? If so I can merge the PR and this will be fixed in the Daily Builds. Thanks.

**Comment by ferdnyc** [comment]
Correction:

https://drive.google.com/file/d/1LIXaUEezQjiganUjr-DIvkn1Wvfc5gxi/view?usp=sharing

...I'm not sure the previous link had the correct permissions.

**Comment by SebO-L** [comment]
@ferdnyc 
Hi, sorry it did not solve the issue. In fact, I am not able to reproduce the error, because the patch create another bug.
Openshot is not able to find the blender executable any more. I did two different tries.
At first, I kept the blender binary in the download folder ("Téléchargement" in french) and it was not working. 
Second try, as it seems to be related to ascii encoding I moved the binary in my home folder and create a symbolic link in /usr/local/bin. But, the result is the same, Openshot can't find blender binary.

I Attach openshot-qt.log and the blender -v command result
Regards, 


[openshot-qt.log](https://github.com/OpenShot/openshot-qt/files/4540508/openshot-qt.log)
[blender_verbose.txt](https://github.com/OpenShot/openshot-qt/files/4540516/blender_verbose.txt)

![screenshot](https://user-images.githubusercontent.com/59599379/80389911-02be4d80-88ac-11ea-9fba-63e3734a5f90.png)

**Comment by ferdnyc** [comment]
Thanks for testing! I'm guessing this means it failed to run the Blender version check with the new code, the logs should tell me why.

**Comment by ferdnyc** [comment]
\*sigh\* Or not, because there's no error logged at all. But it's pretty clear what the issue is. Because we can't pass `encoding='utf-8'` to `subprocess.Popen()` until Python 3.6, I tried to work around that by doing a `.reconfigure(encoding='utf-8')` on the stream after it was opened. Problem is, `.reconfigure()` isn't supported until Python _3.**7**_! So, it's back to the drawing board.

...Stinkin' Python 3.4. :angry:

**Comment by SebO-L** [comment]
OK, 
BTW, as a workaround, I moved the blender binary in a folder without french character and it's working.(with OpenShot-v2.5.1-dev2-1587624156-15c3efb7-26b7db94-x86_64.AppImage)

So I don't know is it's useful for you to speed too much time on that bug nor to wait for the new python version.

Regards,

**Comment by ferdnyc** [comment]
@SebO-L Great, I was going to suggest exactly that workaround.

But, no, we're stuck with Python 3.4 for the foreseeable future, so this has to be made to work with it — however convoluted and backwards that ends up forcing the code to be.

**Comment by ferdnyc** [comment]
Ohhh-kay. This is an AppImage with #3416 integrated:

https://drive.google.com/file/d/1ipXiuDpUxz057jGtapgDh5XyvxtoZRqE/view?usp=sharing

I've already tested it myself, this time, but if you'd like to as well that'd be great. I'm pretty confident in the fix this time, though, as I successfully rendered an animated title using a `/var/tmp/Fjörd/blender` symlink as my Blender path. :wink: Once #3416 is merged the Daily Builds will also include the fix. 

Thanks for the helpful report(s)!
